src/invidious/routes/channels.cr

```diff
@@ -187,6 +187,38 @@ module Invidious::Routes::Channels
     templated "community"
   end
 
+  def self.post(env)
+    # /post/{postId}
+    id = env.params.url["id"].to_s
+    ucid = env.params.query["ucid"]?
+
+    prefs = env.get("preferences").as(Preferences)
+
+    locale = prefs.locale
+    region = env.params.query["region"]? || prefs.region
+
+    thin_mode = env.params.query["thin_mode"]? || prefs.thin_mode
+    thin_mode = thin_mode == "true"
+
+    client_config = YoutubeAPI::ClientConfig.new(region: region)
+
+    if !ucid.nil?
+      ucid = ucid.to_s
+      post_response = fetch_channel_community_post(ucid, id, locale, "json", thin_mode)
+    else
+      # resolve the url to get the author's UCID
+      response = YoutubeAPI.resolve_url("https://www.youtube.com/post/#{id}")
+      return error_template(400, "Invalid post ID") if response["error"]?
+
+      ucid = response.dig("endpoint", "browseEndpoint", "browseId").to_s
+      params = response.dig("endpoint", "browseEndpoint", "params").to_s
```

Comment:
When it comes to the `JSON::Any` type `#to_s` doesn't actually do what you think it does. It instead converts the representation of the object to a string :P

See: https://crystal-lang.org/api/1.9.2/JSON/Any.html#to_s%28io%3AIO%29%3ANil-instance-method

In this case, it works because—well—the underlying object _is_ just a string. It's not a good way to go about it though. 

Please use `#as_s` instead.

Reply:
As a side note I know a bunch of current Invidious code also uses `#to_s` instead of `#as_s`. It's really not ideal and something to note for the future.